Capstone v6 update (without Mips)

Rot127 opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)


**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

Bumps Capstone version to newest Capstone `next` (beyond first v6-Alpha1).

Changes:

- Fixes leaks
- Fixes build and change to AArch64 and SystemZ compatibility headers.
- Marks M68k test as broken (see commit message).
- Fix AArch64 and SystemZ tests
- Handle `op.size == 0` for x86 IL ops

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

All green

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

Rot127 committed:
Handle ldr case without base register.

Rot127 committed:
Fix memory leaks

Rot127 committed:
Add Capstone v6 and bump next to newest commit.

Rot127 committed:
Fix tests.

Rot127 committed:
Change to AArch64 compatibility header

Rot127 committed:
Fix more CSv6 changes in naming

Rot127 committed:
Fix CS v6 Mips related build issues.

Rot127 committed:
Fix CS v6 SystemZ related build issues.

Rot127 committed:
Fix SystemZ analysis tests after CSv6 update

Rot127 committed:
Mark all failing MIPS tests as broken.

Rot127 committed:
Fix Sparc tests.

Rot127 committed:
Mark M68k tesst broken.

This test is broken. The reason is that Rizin can't distinguish
CPU models. Each M68k model has a different address mask (32-24 bits).
It defaults to one which has only 24bits.
Hence, the subi.l instruction decodes to
'subi.l 0x15b119, d0' instead of 'subi.l 0x8015b119, d0'
Should be fixable with RzArch (if anyone is still intereses).
See: https://github.com/rizinorg/rizin/issues/4334

Rot127 committed:
Handle op.size == 0 case.

Rot127 committed:
Fix TriCore tests.

The tests broke due to https://github.com/capstone-engine/capstone/pull/2212.

Rot127 committed:
Run clang-format

XVilka commented on test/db/formats/elf/m68k:
```diff
@@ -1,6 +1,14 @@
+# It defaults to one which has only 24bits.
+# Hence, the subi.l instruction decodes to
+# `subi.l 0x15b119, d0` instead of `subi.l 0x8015b119, d0`
+# Should be fixable with RzArch (if anyone is still intereses).
```
Open an issue instead and link to it here.

wargio replied:
let me handle this, i think it can be resolved in the same way as for mips, because it looks like the exact same issue i faced.

XVilka commented on doc/PACKAGERS.md:
```diff
@@ -66,7 +66,7 @@ version of Capstone be dynamically linked at runtime. To do this, use the
 
 You can override the version of Capstone Rizin will use by setting
-`use_capstone_version` to one of `v4`, `v5` or `next`.
+`use_capstone_version` to one of `v5`, `v6` or `next`.
```
Keep `v4` for now, until it's actually removed from the Rizin codebase

wargio commented on .github/workflows/ci.yml:
```diff
@@ -159,6 +159,16 @@ jobs:
+            enabled: ${{ (github.event_name != 'pull_request' || contains(github.head_ref, 'capstone')) && needs.changes.outputs.edited == 'true' }}
+            timeout: 45
+            cflags: "-Wno-cpp"
+            allow_failure: true
```
this should be false.

wargio commented on test/db/analysis/vars:
```diff
@@ -1141,28 +1141,7 @@ EXPECT=<<EOF
-|       |   0x8000054a      mov   d15, #1
-|       `-> 0x8000054c      mov   d2, d15
-\           0x8000054e      ret
+\       |   0x80000514      ld.a  a15, [a14]#-0x18
```
This looks like a regression. before it was reaching the `ret`, but now it stops at `ld.a`

Rot127 replied:
Yes, it does. But The call addresses seem correct (at least did @imbillow reviewed the corresponding Capstone PR). I don't know enough about TriCore. Maybe they compile overlapping functions.

b1llow replied:
See my reply in mattermost.

Rot127 committed:
Revert "Fix TriCore tests."

This reverts commit 8b55655a941687f4c4fede6a694edfac3f00d949.

Rot127 committed:
Bump Caspstone to newest next.

Rot127 committed:
Fix tests.

With https://github.com/capstone-engine/capstone/pull/2504 two false positives and a new function is discovered.

Rot127 committed:
Add Capstone v4 again to docs

Rot127 committed:
Mark Mips tests broken

Rot127 committed:
Don't allow v6 CI job for CS v6 to fail.

Rot127 committed:
Build Capstone next with SystemZ and print warning if it couldn't init the handle.

Rot127 committed:
Bump Ubuntu runner to 24.04.

This is an attempt to fix a reported false positive warning of GCC 12
which is not reported by GCC 14.

Rot127 committed:
Fix TriCore asm tests after https://github.com/capstone-engine/capstone/pull/2504

Rot127 committed:
Revert "Bump Ubuntu runner to 24.04."

This reverts commit 7949b02d902463f76953d26738e562695fbef24d.

Rot127 committed:
Add link to issue

Rot127 commented:
Fix for build issue: https://github.com/capstone-engine/capstone/pull/2522

XVilka commented:
@Rot127 please update the Capstone commit and rebase one more time

Rot127 committed:
Bump to newest Capstone next

XVilka commented:
@Rot127 please fix the build with older Capstone versions too:
```
gcc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -Isubprojects/rzspp -I../subprojects/rzspp -I/usr/include/capstone -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -DUSE_SYS_CAPSTONE -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil32.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil32.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil32.c.o -c ../librz/arch/isa/arm/arm_esil32.c
In file included from ../librz/arch/isa/arm/arm_esil32.c:10:
../librz/arch/isa/arm/arm_cs.h:24:67: error: unknown type name ‘ARM64CC_CondCode’
   24 | RZ_IPI const char *rz_arm64_cs_esil_prefix_cond(RzAnalysisOp *op, ARM64CC_CondCode cond_type);
      |                                                                   ^~~~~~~~~~~~~~~~
[1481/2211] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil64.c.o
FAILED: librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil64.c.o 
gcc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -Isubprojects/rzspp -I../subprojects/rzspp -I/usr/include/capstone -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -DUSE_SYS_CAPSTONE -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil64.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil64.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_esil64.c.o -c ../librz/arch/isa/arm/arm_esil64.c
In file included from ../librz/arch/isa/arm/arm_esil64.c:11:
../librz/arch/isa/arm/arm_cs.h:24:67: error: unknown type name ‘ARM64CC_CondCode’
   24 | RZ_IPI const char *rz_arm64_cs_esil_prefix_cond(RzAnalysisOp *op, ARM64CC_CondCode cond_type);
      |                                                                   ^~~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:18:67: error: unknown type name ‘ARM64CC_CondCode’
   18 | RZ_IPI const char *rz_arm64_cs_esil_prefix_cond(RzAnalysisOp *op, ARM64CC_CondCode cond_type) {
      |                                                                   ^~~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:132:28: error: unknown type name ‘aarch64_extender’; did you mean ‘arm64_extender’?
  132 | static int decode_sign_ext(aarch64_extender extender) {
      |                            ^~~~~~~~~~~~~~~~
      |                            arm64_extender
../librz/arch/isa/arm/arm_esil64.c:156:36: error: unknown type name ‘aarch64_shifter’; did you mean ‘arm64_shifter’?
  156 | static const char *decode_shift_64(aarch64_shifter shift) {
      |                                    ^~~~~~~~~~~~~~~
      |                                    arm64_shifter
../librz/arch/isa/arm/arm_esil64.c: In function ‘shifted_reg64_append’:
../librz/arch/isa/arm/arm_esil64.c:154:18: error: implicit declaration of function ‘decode_sign_ext’ [-Werror=implicit-function-declaration]
  154 | #define EXT64(x) decode_sign_ext(insn->detail->arm64.operands[x].ext)
      |                  ^~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:207:23: note: in expansion of macro ‘EXT64’
  207 |         int signext = EXT64(n);
      |                       ^~~~~
../librz/arch/isa/arm/arm_esil64.c:180:27: error: implicit declaration of function ‘decode_shift_64’ [-Werror=implicit-function-declaration]
  180 | #define DECODE_SHIFT64(x) decode_shift_64(insn->detail->arm64.operands[x].shift.type)
      |                           ^~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:219:100: note: in expansion of macro ��DECODE_SHIFT64’
  219 |                                 rz_strbuf_appendf(sb, "%d,%d,%s,~,%s", LSHIFT2_64(n), signext, rn, DECODE_SHIFT64(n));
      |                                                                                                    ^~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:219:68: error: format ‘%s’ expects argument of type ‘char *’, but argument 6 has type ‘int’ [-Werror=format=]
  219 |                                 rz_strbuf_appendf(sb, "%d,%d,%s,~,%s", LSHIFT2_64(n), signext, rn, DECODE_SHIFT64(n));
      |                                                                   ~^
      |                                                                    |
      |                                                                    char *
      |                                                                   %d
../librz/arch/isa/arm/arm_esil64.c:221:63: error: format ‘%s’ expects argument of type ‘char *’, but argument 5 has type ‘int’ [-Werror=format=]
  221 |                                 rz_strbuf_appendf(sb, "%d,%s,%s", LSHIFT2_64(n), rn, DECODE_SHIFT64(n));
      |                                                              ~^
      |                                                               |
      |                                                               char *
      |                                                              %d
../librz/arch/isa/arm/arm_esil64.c:231:68: error: format ‘%s’ expects argument of type ‘char *’, but argument 6 has type ‘int’ [-Werror=format=]
  231 |                                 rz_strbuf_appendf(sb, "%d,%d,%s,~,%s,1,%d,%s,~,<<<,1,&,?{,%" PFMT64u ",}{,0,},|",
      |                                                                   ~^
      |                                                                    |
      |                                                                    char *
      |                                                                   %d
../librz/arch/isa/arm/arm_esil64.c:234:63: error: format ‘%s’ expects argument of type ‘char *’, but argument 5 has type ‘int’ [-Werror=format=]
  234 |                                 rz_strbuf_appendf(sb, "%d,%s,%s,1,%s,<<<,1,&,?{,%" PFMT64u ",}{,0,},|",
      |                                                              ~^
      |                                                               |
      |                                                               char *
      |                                                              %d
../librz/arch/isa/arm/arm_esil64.c: In function ‘rz_arm_cs_analysis_op_64_esil’:
../librz/arch/isa/arm/arm_esil64.c:404:19: error: implicit declaration of function ‘rz_arm64_cs_esil_prefix_cond’; did you mean ‘rz_arm32_cs_esil_prefix_cond’? [-Werror=implicit-function-declaration]
  404 |         postfix = rz_arm64_cs_esil_prefix_cond(op, insn->detail->arm64.cc);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                   rz_arm32_cs_esil_prefix_cond
../librz/arch/isa/arm/arm_esil64.c:404:17: error: assignment to ‘const char *’ from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
  404 |         postfix = rz_arm64_cs_esil_prefix_cond(op, insn->detail->arm64.cc);
      |                 ^
../librz/arch/isa/arm/arm_esil64.c:714:70: error: format ‘%s’ expects argument of type ‘char *’, but argument 6 has type ‘int’ [-Werror=format=]
  714 |                                         rz_strbuf_appendf(&op->esil, "%s,%d,%" PFMT64d ",%s,+",
      |                                                                      ^~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:714:91: note: format string is defined here
  714 |                                         rz_strbuf_appendf(&op->esil, "%s,%d,%" PFMT64d ",%s,+",
      |                                                                                          ~^
      |                                                                                           |
      |                                                                                           char *
      |                                                                                          %d
../librz/arch/isa/arm/arm_esil64.c:815:70: error: format ‘%s’ expects argument of type ‘char *’, but argument 7 has type ‘int’ [-Werror=format=]
  815 |                                         rz_strbuf_appendf(&op->esil, "%d,%s,%d,%" PFMT64d ",%s",
      |                                                                      ^~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:815:94: note: format string is defined here
  815 |                                         rz_strbuf_appendf(&op->esil, "%d,%s,%d,%" PFMT64d ",%s",
      |                                                                                             ~^
      |                                                                                              |
      |                                                                                              char *
      |                                                                                             %d
../librz/arch/isa/arm/arm_esil64.c:991:70: error: format ‘%s’ expects argument of type ‘char *’, but argument 7 has type ‘int’ [-Werror=format=]
  991 |                                         rz_strbuf_appendf(&op->esil, "%s,%s,%d,%" PFMT64d ",%s,+",
      |                                                                      ^~~~~~~~~~~~
../librz/arch/isa/arm/arm_esil64.c:991:94: note: format string is defined here
  991 |                                         rz_strbuf_appendf(&op->esil, "%s,%s,%d,%" PFMT64d ",%s,+",
      |                                                                                             ~^
      |                                                                                              |
      |                                                                                              char *
      |                                                                                             %d
cc1: all warnings being treated as errors
[1482/2211] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il64.c.o
FAILED: librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il64.c.o 
gcc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -Isubprojects/rzspp -I../subprojects/rzspp -I/usr/include/capstone -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -DUSE_SYS_CAPSTONE -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il64.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il64.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il64.c.o -c ../librz/arch/isa/arm/arm_il64.c
In file included from ../librz/arch/isa/arm/arm_il64.c:11:
../librz/arch/isa/arm/arm_cs.h:24:67: error: unknown type name ‘ARM64CC_CondCode’
   24 | RZ_IPI const char *rz_arm64_cs_esil_prefix_cond(RzAnalysisOp *op, ARM64CC_CondCode cond_type);
      |                                                                   ^~~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_il64.c:42:25: error: unknown type name ‘ARM64CC_CondCode’
   42 | static RzILOpBool *cond(ARM64CC_CondCode c) {
      |                         ^~~~~~~~~~~~~~~~
../librz/arch/isa/arm/arm_il64.c: In function ‘branch’:
../librz/arch/isa/arm/arm_il64.c:637:25: error: implicit declaration of function ‘cond’ [-Werror=implicit-function-declaration]
  637 |         RzILOpBool *c = cond(insn->detail->arm64.cc);
      |                         ^~~~
../librz/arch/isa/arm/arm_il64.c:637:25: error: initialization of ‘RzILOpBool *’ {aka ‘struct rz_il_op_pure_t *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
../librz/arch/isa/arm/arm_il64.c: In function ‘cmp’:
../librz/arch/isa/arm/arm_il64.c:890:25: error: initialization of ‘RzILOpBool *’ {aka ‘struct rz_il_op_pure_t *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
  890 |         RzILOpBool *c = cond(insn->detail->arm64.cc);
      |                         ^~~~
../librz/arch/isa/arm/arm_il64.c: In function ‘csinc’:
../librz/arch/isa/arm/arm_il64.c:924:25: error: initialization of ‘RzILOpBool *’ {aka ‘struct rz_il_op_pure_t *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
  924 |         RzILOpBool *c = cond(insn->detail->arm64.cc);
      |                         ^~~~
../librz/arch/isa/arm/arm_il64.c: In function ‘cset’:
../librz/arch/isa/arm/arm_il64.c:1004:11: error: assignment to ‘RzILOpBool *’ {aka ‘struct rz_il_op_pure_t *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
 1004 |         c = cond(insn->detail->arm64.cc);
      |           ^
cc1: all warnings being treated as errors
gcc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -Isubprojects/rzspp -I../subprojects/rzspp -I/usr/include/capstone -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -DUSE_SYS_CAPSTONE -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -c ../librz/arch/isa/arm/arm_il32.c
In file included from ../librz/arch/isa/arm/arm_il32.c:12:
../librz/arch/isa/arm/arm_cs.h:24:67: error: unknown type name ‘ARM64CC_CondCode’
   24 | RZ_IPI const char *rz_arm64_cs_esil_prefix_cond(RzAnalysisOp *op, ARM64CC_CondCode cond_type);
      |                                                                   ^~~~~~~~~~~~~~~~
[1485/2211] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_8051_8051_ass.c.o
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.
```

XVilka commented:
@Rot127 see also
```
cc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -I../subprojects/capstone-next/include -I../subprojects/capstone-next/include/capstone -Isubprojects/rzspp -I../subprojects/rzspp -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 --std=gnu99 -Werror=sizeof-pointer-memaccess -Wimplicit-fallthrough=3 -fvisibility=hidden -DRZ_PLUGIN_INCORE=1 -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -c ../librz/arch/isa/arm/arm_il32.c
../librz/arch/isa/arm/arm_il32.c:8:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
    8 | #pragma GCC diagnostic ignored "-Wenum-conversion"
      |                                ^~~~~~~~~~~~~~~~~~~
In file included from ../librz/arch/isa/arm/arm_il32.c:12:
../librz/arch/isa/arm/arm_cs.h:9:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
    9 | #pragma GCC diagnostic ignored "-Wenum-conversion"
      |                                ^~~~~~~~~~~~~~~~~~~
In file included from ../librz/arch/isa/arm/arm_il32.c:13:
../librz/arch/isa/arm/arm_accessors32.h:11:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
   11 | #pragma GCC diagnostic ignored "-Wenum-conversion"
      |                                ^~~~~~~~~~~~~~~~~~~
```

Rot127 commented:
Damn, I thought the whole time I am on the v5/v4 building branch.
Anyways, here is the patch: https://github.com/capstone-engine/capstone/pull/2525
